test/connectors/CustomsFinancialsApiConnectorSpec.scala

```diff
@@ -330,6 +330,96 @@ class CustomsFinancialsApiConnectorSpec extends SpecBase {
     }
   }
 
+  "postCashAccountStatements" must {
+    "return success when calling the correct URL" in new Setup {
+
+      val expectedUrl = "apiEndpointUrl/accounts/cashaccountstatementrequest/v1"
+
+      when(requestBuilder.withBody(any())(any(), any(), any())).thenReturn(requestBuilder)
+
+      when(requestBuilder.execute(any[HttpReads[CashTransactions]], any[ExecutionContext]))
+        .thenReturn(Future.successful(accResponse))
+
+      when(mockHttpClient.post(any[URL]())(any())).thenReturn(requestBuilder)
+
+      val appWithMocks: Application = application
+        .overrides(
+          bind[HttpClientV2].toInstance(mockHttpClient),
+          bind[RequestBuilder].toInstance(requestBuilder)
+        ).build()
+
+      running(appWithMocks) {
+        val result = await(connector(appWithMocks).postCashAccountStatements(
+          "eori","can", fromDate, toDate))
+
+        result mustBe Right(accResponse)
+      }
+    }
+
+    "return failure when backend post fails" in new Setup {
+
+      private val responseCode: Int = 500
+
+      when(requestBuilder.withBody(any())(any(), any(), any())).thenReturn(requestBuilder)
+
+      when(requestBuilder.execute(any[HttpReads[Seq[CashDailyStatement]]], any[ExecutionContext]))
+        .thenReturn(Future.failed(new HttpException("It's broken", responseCode)))
+
+      when(mockHttpClient.post(any())(any())).thenReturn(requestBuilder)
+
+      val appWithMocks: Application = application
+        .overrides(bind[HttpClientV2].toInstance(mockHttpClient)).build()
+
+      running(appWithMocks) {
+        connector(appWithMocks).postCashAccountStatements("eori", "can", fromDate, toDate).map {
+          _ mustBe Left(UnknownException)
+        }
+      }
+    }
+
+    "return ErrorResponse when the backend POST fails with REQUEST_ENTITY_TOO_LARGE" in new Setup {
+
+      when(requestBuilder.withBody(any())(any(), any(), any())).thenReturn(requestBuilder)
+
+      when(requestBuilder.execute(any[HttpReads[Seq[CashDailyStatement]]], any[ExecutionContext]))
+
+        .thenReturn(Future.failed(UpstreamErrorResponse("Error occurred", REQUEST_ENTITY_TOO_LARGE)))
+      when(mockHttpClient.post(any())(any())).thenReturn(requestBuilder)
+
+      val appWithMocks: Application = application
+        .overrides(
+          bind[HttpClientV2].toInstance(mockHttpClient)
+        ).build()
+
+      running(appWithMocks) {
+        connector(appWithMocks).retrieveCashTransactionsDetail("can", fromDate, toDate).map {
+          _ mustBe Left(TooManyTransactionsRequested)
+        }
+      }
+    }
+
+    "return ErrorResponse when the backend POST fails with NOT_FOUND" in new Setup {
```

Comment:
There is no condition for this in the method, there are only below four failure scenarios hence update the test accordingly (including error response in assertion) for the correct failure

BAD_REQUEST, INTERNAL_SERVER_ERROR, SERVICE_UNAVAILABLE and rest of them